### PR TITLE
Add Coverity scan

### PR DESCRIPTION
**Pull request by omalyshe**
### Description 
This adds the Coverity scan for the project https://scan.coverity.com/projects/oneapi-src-onetbb. 
It also add the badge to README file.

There is a limitation for the # of runs per day/week. For oneTBB project the following is applicable: _Up to 21 builds per week, with a maximum of 3 builds per day, for projects with 100K to 500K lines of code_
So, we'll run only on merge to master.

P.S. The existing project works with the repo migrated to uxlfoundation. 
Successful run - https://github.com/uxlfoundation/oneTBB/actions/runs/12353089723/job/34471460976?pr=1455 

Fixes # - _issue number(s) if exists_

### Type of change

_Choose one or multiple, leave empty if none of the other choices apply_

_Add a respective label(s) to PR if you have permissions_

- [ ] bug fix - _change that fixes an issue_
- [ ] new feature - _change that adds functionality_
- [ ] tests - _change in tests_
- [X] infrastructure - _change in infrastructure and CI_
- [ ] documentation - _documentation update_

### Tests

- [ ] added - _required for new features and some bug fixes_
- [X] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [X] not needed

### Breaks backward compatibility
- [ ] Yes
- [X] No
- [ ] Unknown

### Notify the following users
_List users with `@` to send notifications_

### CODE REVIEW

File: .github/workflows/coverity.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  coverity_linux:
+    name: Coverity Linux
+    runs-on: [ubuntu-latest]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Download Linux 64 Coverity Tool
+        run: |
+          curl https://scan.coverity.com/download/cxx/linux64 --output ${GITHUB_WORKSPACE}/cov-linux64-tool.tar.gz \
+            --data "token=${{secrets.COVERITY_TOKEN}}&project=${{env.COVERITY_PROJECT}}"
+          ls -l ${GITHUB_WORKSPACE}
```
Comment (isaevil) [inline .github/workflows/coverity.yml:37]: Seems like was used for debug.
```suggestion
```

Comment (omalyshe) [reply .github/workflows/coverity.yml]: removed

File: .github/workflows/coverity.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+# Only run on push to master branch
+  push:
+    branches: [master]
+
+permissions: read-all
+
+env:
+  BUILD_CONCURRENCY: 4
+  COVERITY_PROJECT: oneapi-src%2FoneTBB
```
Comment (isaevil) [inline .github/workflows/coverity.yml:27]: Is it possible to change coverity project name from `oneapi-src` to `uxlfoundation`?

**Review by isaevil — COMMENTED** [review]
Looks good to me. One small question though.

File: .github/workflows/coverity.yml (hunk 2 above)
Comment (omalyshe) [reply .github/workflows/coverity.yml:27]: > Is it possible to change coverity project name from `oneapi-src` to `uxlfoundation`?

From what I found in the documentation - no, it is not possible to change the project name. Only create the new one.

**Review by isaevil — APPROVED** [review]

**Pull request merged**